LE Audio Split and update broadcast sink API

Thalley opened the pull request:
Updates the broadcast sink API to be completely separate from the connected audio API, and updates a few function names.

Thalley committed:
Bluetooth: Audio: Add _sink infix to broadcast sink functions

The broadcast sink functions will now have a _sink_ infix
to be more similar to the broadcast_source functions.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add check for broadcast sink in bt_audio_chan_send

Add check for broadcast sink in bt_audio_chan_send and stop
it from sending.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Rename broadcast sink release to delete

Rename bt_audio_broadcast_sink_release to
bt_audio_broadcast_sink_delete as that is more descriptive of what
the function does, as well as matches better the
broadcast source API.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Move sink stop to own function

Move the functionality of stopping the sink to its own
function. This is to follow the broadcast API.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Fix sink chan state issues

The channel were not put in or testet for the
BT_AUDIO_CHAN_CONFIGURED state, which made the state
transition illegal once the BIS "connected".

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: test: Add broadcast_sink_disconnect test

Add a test case where the broadcast sink initiates the
deletion of the sink.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo commented on include/bluetooth/audio.h:
```diff
@@ -921,7 +921,7 @@ int bt_audio_chan_stop(struct bt_audio_chan *chan);
  *  source channel.
  *
  *  Broadcast sink channels shall be released using
```
It sound weird to say that they shall be released using a delete function.
Propose to use same wording as for source (just below): "cannot be released, but can be deleted [...]"

asbjornsabo replied:
A missing "d" again ... 
(Nitpicking, I know.  But it looks better without spelling errors.)

Thalley replied:
Odd that I keep making that typo... Oh well, it's fixed :D

asbjornsabo commented on include/bluetooth/audio.h:
```diff
@@ -1097,15 +1097,15 @@ int bt_audio_broadcast_sink_sync(struct bt_audio_broadcast_sink *sink,
  *
  *  Once a broadcast sink has been allocated after the pa_synced callback,
- *  it can be released using this function. If the sink has synchronized to any
+ *  it can be delete using this function. If the sink has synchronized to any
```
"deleted"

asbjornsabo commented on subsys/bluetooth/host/audio/chan.c:
```diff
@@ -734,7 +734,7 @@ int bt_audio_chan_release(struct bt_audio_chan *chan, bool cache)
 		return -EINVAL;
 	} else if (bt_audio_ep_is_broadcast_snk(chan->ep)) {
-		BT_DBG("Broadcast sink shall be released with bt_audio_broadcast_sink_release");
+		BT_DBG("Broadcast sink shall be released with bt_audio_broadcast_sink_delete");
```
Same release/delete.  Suggest to reuse wording for source here as well.

asbjornsabo commented on subsys/bluetooth/host/audio/chan.c:
```diff
@@ -2477,10 +2468,41 @@ int bt_audio_broadcast_sink_sync(struct bt_audio_broadcast_sink *sink,
+	chan = sink->chan;
+
+	if (chan->state != BT_AUDIO_CHAN_STREAMING ||
+	    chan->state != BT_AUDIO_CHAN_CONFIGURED) {
```
Should be AND ("&&") - the state is going to be un-equal to at least one of them, so this test will always be true.

Thalley replied:
Thanks!

asbjornsabo commented on tests/bluetooth/bsim_bt/bsim_test_audio/src/broadcast_sink_test.c:
```diff
@@ -127,13 +157,53 @@ static void test_main(void)
+		FAIL("Unable to delete sink: %d", err);
+		return;
+	}
+	/* No "sync lost" event is generated when we initialie the disconnect */
```
Nitpicking: initialize - missing "z"

Thalley committed:
Bluetooth: Audio: Fix broadcast sink data path issue

The ISO qos was incorrectly set, and was not actually
used for the ISO API, thus failing the iso data path.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo reviewed (approved):
One minor spelling correction left, otherwise LGTM
